src/sonic-yang-models/yang-models/sonic-events-host.yang

```diff
@@ -179,5 +179,113 @@ module sonic-events-host {
+                    }
+                }
+                description "Container name";
+            }
+
+
+            leaf mem_usage {
+                type uint64;
+                description "Memory usage of process";
+            }
+
+            leaf threshold {
+                type uint64;
+                description "Threshold value of process";
+            }
+
+            uses evtcmn:sonic-events-cmn;
+        }
+
+        container process-exited-unexpectedly {
+            evtcmn:EVENT_SEVERITY_2;
+
+            description "
+                Declares an event in which a critical process exits unexpectedly.";
+
+            leaf process_name {
+                type string {
+                    pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,31})' {
+                        error-message "Invalid process exited unexpectedly process_name.";
+                        error-app-tag process-exited-unexpectedly-invalid-process-name;
+                    }
+                    length 1..32 {
+                        error-message "Invalid length for process exited unexpectedly process_name.";
+                        error-app-tag process-exited-unexpectedly-invalid-process-name-length;
+                    }
+                }
+                description "Name of process that is exiting";
+            }
+
+            leaf ctr_name {
+                type string {
+                    pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,31})' {
+                        error-message "Invalid process exited unexpectedly ctr_name.";
+                        error-app-tag process-exited-unexpectedly-invalid-ctr-name;
+                    }
+                    length 1..32 {
+                        error-message "Invalid length for process exited unexpectedly ctr_name.";
+                        error-app-tag process-exited-unexpectedly-invalid-ctr-name-length;
+                    }
+                }
+                description "Container name";
+            }
+
+            uses evtcmn:sonic-events-cmn;
+        }
+
+        container process-not-running {
+            evtcmn:EVENT_SEVERITY_2;
+
+            description "
+                Declares an event in which a critical process exits unexpectedly.";
+
+            leaf process_name {
+                type string {
+                    pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,31})' {
+                        error-message "Invalid process not running process_name.";
+                        error-app-tag process-not-running-invalid-process-name;
+                    }
+                    length 1..32 {
+                        error-message "Invalid length for process not running process_name.";
+                        error-app-tag process-not-running-invalid-process-name-length;
+                    }
+                }
+                description "Name of process that is not running";
+            }
+
+            leaf ctr_name {
```

Comment:
ctr_name and process_name are used repeatedly, maybe we can add them to sonic-type.yang?

Reply:
Done